### PR TITLE
remove mmap option

### PR DESCRIPTION
**Pull request by arnaudbore**
#531 + @GuillaumeTh answer: "Je dirais que faudrait retirer l'option"

### CODE REVIEW

File: scripts/scil_compute_local_tracking_dev.py [hunk 1]
```diff
@@ -186,7 +176,7 @@ def main():
     logging.debug("Instantiating tracker.")
     tracker = Tracker(propagator, mask, seed_generator, nbr_seeds, min_nbr_pts,
                       max_nbr_pts, max_invalid_dirs, args.compress,
-                      args.nbr_processes, args.save_seeds, mmap_mode,
+                      args.nbr_processes, args.save_seeds, mmap_mode=None,
```
Comment (GuillaumeTh) [inline scripts/scil_compute_local_tracking_dev.py:179]: It is 'r+' not None

File: scripts/scil_compute_local_tracking_dev.py [hunk 2]
```diff
@@ -107,12 +107,6 @@ def _build_arg_parser():
                           "with -nt 1,000,000, \nyou can create tractogram_2 "
                           "with \n--skip 1,000,000.")
 
-    m_g = p.add_argument_group('Memory options')
-    add_processes_arg(m_g)
```
Comment (EmmaRenauld) [inline scripts/scil_compute_local_tracking_dev.py:111]: This line should not be removed (add_processes_arg).

**Comment by EmmaRenauld** [comment]
I have no idea how to review this. Can you explain what you changed in the fetcher?

File: scripts/scil_compute_local_tracking_dev.py [hunk 3]
```diff
@@ -186,7 +176,7 @@ def main():
     logging.debug("Instantiating tracker.")
     tracker = Tracker(propagator, mask, seed_generator, nbr_seeds, min_nbr_pts,
                       max_nbr_pts, max_invalid_dirs, args.compress,
-                      args.nbr_processes, args.save_seeds, mmap_mode,
+                      args.nbr_processes, args.save_seeds, mmap_mode='r+',
```
Comment (EmmaRenauld) [inline scripts/scil_compute_local_tracking_dev.py:179]: In the Tracker, could we improve the description of mmap_mode. Ex: add suggestions, explanations.

**Comment by scil-admin** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-549/3/

File: scripts/scil_compute_local_tracking_dev.py (hunk 3 above)
Comment (arnaudbore) [reply scripts/scil_compute_local_tracking_dev.py:182]: @GuillaumeTh @CHrlS98 ? any suggestions ?

File: scripts/scil_compute_local_tracking_dev.py [hunk 4]
```diff
@@ -186,7 +179,7 @@ def main():
...
```
Comment (CHrlS98) [inline scripts/scil_compute_local_tracking_dev.py:182]: You have positional arguments after your keyword argument and it crashes

File: scripts/scil_compute_local_tracking_dev.py (hunk 3 above)
Comment (CHrlS98) [reply scripts/scil_compute_local_tracking_dev.py:182]: Right now it reads
```
mmap_mode: str
            Memory-mapping mode. One of {None, ‘r+’, ‘r’, ‘w+’, ‘c’}
```
First, @GuillaumeTh and I ran some tests and options 'r' and 'w+' result in a crash, so they shouldn't be in the choices. The descriptions for the remaining mmap mode can be found here https://numpy.org/doc/stable/reference/generated/numpy.memmap.html#numpy.memmap. I'd just copy the corresponding descriptions and say it tells the tracker how to access the data across multiple threads. Maybe we should add a safety check in the `__init__` to validate that memmap is one of {None, 'r+', 'c'}.

**Comment by scil-admin** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-549/4/

File: scripts/scil_compute_local_tracking_dev.py (hunk 3 above)
Comment (arnaudbore) [reply scripts/scil_compute_local_tracking_dev.py:182]: Can I put 'r+' as default mmap_mode for the Tracker ? @CHrlS98 @GuillaumeTh @EmmaRenauld  ?

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

File: scilpy/tracking/tracker.py [hunk 5]
```diff
@@ -49,7 +49,7 @@ def __init__(self, propagator: AbstractPropagator, mask: DataVolume,
             Whether to save the seeds associated to their respective
             streamlines.
         mmap_mode: str
-            Memory-mapping mode. One of {None, ‘r+’, ‘r’, ‘w+’, ‘c’}
+            Memory-mapping mode to access the tracker. One of {None, ‘r+’, ‘c’}
```
Comment (CHrlS98) [inline scilpy/tracking/tracker.py:52]: It is for accessing the tracking data (for example SH coefficients image when tracking on ODF). Something like this?
```suggestion
            Memory-mapping mode. One of {None, 'r+', 'c'}. This value is passed to
            np.load() when loading the raw tracking data from a subprocess.
```

**Review by CHrlS98 — APPROVED** [review]

**Pull request merged**